backend/sirius-web-emf/src/main/java/org/eclipse/sirius/web/emf/services/EditService.java

```diff
@@ -202,7 +202,10 @@ public EditService(IEditingContextEPackageService editingContextEPackageService,
         editingDomain.getCommandStack().execute(createChildCommand);
         Collection<?> result = createChildCommand.getResult();
         if (result.size() == 1) {
-            objectOptional = Optional.of(result.iterator().next());
+            Object child = result.iterator().next();
+            if (child instanceof EObject && EcoreUtil.isAncestor(eObject, (EObject) child)) {
+                objectOptional = Optional.of(child);
+            }
```

Comment:
Shouldn't the child be deleted in case it is not an ancestor?

Reply:
I don't think, because the "child" is a blank EObject (in our case a blank NodeStyle) which has been provided by AdapterFactoryEditingDomain#getNewChildDescriptors (NodeDescriptionItemprovider#collectNewChildDescriptor) and held by the CommandParameter.
Then, the "child" is given to the CreateChildCommand which only adds, sets, or does nothing whether the feature is multi-valued, single-valued, or already set for a single-valued feature.

Reply:
So, if we have to delete the child, we also need to delete other potential "children" provided by AdapterFactoryEditingDomain#getNewChildDescriptors.

Reply:
So you are saying that the child is not in the resource at all?

Reply:
If the create child command did nothing, `child.eContainer() == null`, so in that case, yes, the child is not in a resource, and thus, we should return an ErrorPayload.
If the create child command did add or did set (depending on the feature bounds), `child.eContainer() == eObject`, meaning the child has been "created", and thus, we should return a CreateChildSuccessPayload.